### PR TITLE
fix(modal): fixed centering. Collision happened because SUI sarted us…

### LINKED ISSUE #244: Modal is no longer centered after update to 12.2.0
After updating to SUIR 0.82 and octagon 12.2.0, the `Modal` is no longer centered horizontally and sometimes appears partially offscreen, depending on window size.

### PR DESCRIPTION
**Pull request by andrewleyva**
Collision happened because SUI started using Flexbox to center and our overrides weren't expecting that. This fixes it.

# problem statement

Modal not centered. Close button not positioned properly. This wasn't caught initially because example didn't include the default modal, but only fullscreen (which worked as expected).

# solution

Adjust margins to center modal.
Adjust positioning for close button.
Add Default example.

Fixes #244

### CODE REVIEW

**Comment by andrewleyva** [comment]
The override is because a negative margin was being applied to the modal. The modal was positioned using `left: 50%` and a negative margin equal to half of the modal width. (I expected `transform: translateX(-50%)`). We no longer need a negative margin.

modal-before
modal-after

File: src/semantic-ui-theme/themes/tripwire/modules/modal.overrides [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+// SUIR is now using Flexbox to center the modal. A bug was introducted in SUICSS 2.3
+// This overrides the bug
+
+.ui.modal {
+	margin: 1em auto !important;
+}
+
+.ui.modal > .close {
+	top: @innerCloseTop !important;
+  right: @innerCloseRight !important;
```
Comment (youngbob) [inline src/semantic-ui-theme/themes/tripwire/modules/modal.overrides]: Looks like mixing tabs and spaces here.

**Comment by ggascoigne** [comment]
Ah, OK.  I'd expected to see some code deleted, but if that's not the case that all sorts of makes sense.  I just hate seeing !important anywhere.

File: src/semantic-ui-theme/themes/tripwire/modules/modal.overrides (hunk 1 above)
Comment (andrewleyva) [reply src/semantic-ui-theme/themes/tripwire/modules/modal.overrides]: Oops. I thought my IDE converted tabs to space automatically. Pushing up the fix now.

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereviewmqnkapeypofzlda.surge.sh](http://surgereviewmqnkapeypofzlda.surge.sh)

File: src/components/suir/Modal.examples.md [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+    <Modal trigger={<Button>Show Modal</Button>} closeIcon='close'>
+    <Modal.Header>Select a Photo</Modal.Header>
+    <Modal.Content>
+      <Modal.Description>
+        <Header>Default Profile Image</Header>
+        <p>We've found the following gravatar image associated with your e-mail address.</p>
+        <p>Is it okay to use this photo?</p>
+      </Modal.Description>
+    </Modal.Content>
+    </Modal>
```
Comment (cdaringe) [inline src/components/suir/Modal.examples.md]: i converted all of the examples to use the triple backtick.  mind it we did so here too?  also, closing modal indent is one too many in

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereviewzobkofkrshpolda.surge.sh](http://surgereviewzobkofkrshpolda.surge.sh)

File: src/semantic-ui-theme/themes/tripwire/modules/modal.overrides [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+// SUIR is now using Flexbox to center the modal. A bug was introducted in SUICSS 2.3
+// This overrides the bug
+
+.ui.modal {
+  margin: 1em auto;
+}
+
+.ui.modal > .close {
+  top: @innerCloseTop;
+  right: @innerCloseRight;
```
Comment (youngbob) [inline src/semantic-ui-theme/themes/tripwire/modules/modal.overrides]: It looks like you removed all of the `!important` when converting tabs to spaces. Was that intentional? Does this still work without `!important`?

Comment (andrewleyva) [reply src/semantic-ui-theme/themes/tripwire/modules/modal.overrides]: Yes, it works without `! important `... shouldn't have been there in the first place. Was lazy and sloppy.

**Pull request merged**